sssd_test_framework/roles/ad.py

```diff
@@ -1490,4 +1500,309 @@ def __get_info(
+    def get(self, attrs: list[str]) -> dict[str, list[str]]:
+        """Get Group Policy object.
+
+        :param attrs: If set, only requested attributes are returned, defaults to None
+        :type attrs: list[str] | None, optional
+        :return: Dictionary with attribute name as a key.
+        :rtype: dict[str, list[str]]
+        """
+        cmd = self._exec("Get-GPO", self.cli.args(self._name), format_with="Format-List")
+        return attrs_parse(cmd.stdout_lines, attrs)
+
+    def delete(self) -> None:
+        """
+        Delete Group Policy object.
+        """
+        args: CLIBuilderArgs = {"Confirm": (self.cli.option.SWITCH, False)}
+        self._exec("Remove-GPO", self.cli.args(**self._name, **args, quote_value=True))
+
+
+class GPKey(GPObject):
+    """Group Policy registry management."""
+
+    def __init__(
+        self,
+        role: AD,
+        name: str,
+        key: (
+            str | None
+        ) = "Computer Configuration\
+        Windows Settings\
+        Security Settings\
+        Local Policies\
+        User Rights Assignment",
+    ) -> None:
+        """
+        :param role: AD role object.
+        :type role: AD
+        :param name: Group policy name.
+        :type name: str
+        :param key; Key path.
+        :type key: str
+
+         Allow log on locally    SeInteractiveLogonRight
+         Allow log on through Remote Desktop Services    SeRemoteInteractiveLogonRight
+
+         Deny log on locally SeDenyInteractiveLogonRight
+         Deny log on through Remote Desktop Services SeDenyRemoteInteractiveLogonRight
+        """
+        super().__init__(role, name)
+
+        self.key: str = key
+
+    def _set(
+        self,
+        value_name: str,
+        value: str,
+        context: str | None = "user",
+        value_type: str | None = "String",
+        action: str | None = "Update",
+    ) -> None:
+        """
+        Set the registry values for the Group Policy.
+
+        :param value_name: Name of the value.
+        :type value_name: str
+        :param value: Value of the value name.
+        :type value: str
+        :param context: Context, defaults to "user"
+        :type context: str, optional
+        :param value_type: Value type, defaults to "string"
+        :type value_type: str, optional
+        :param action: Action to be performed, defaults to  "update"
+        :type action: str, optional
+
+        """
+        params: CLIBuilderArgs = {
+            "Name": self._name,
+            "Context": (self.cli.option.VALUE, context),
+            "Key": (self.cli.option.VALUE, self.key),
+            "ValueName": (self.cli.option.VALUE, self.key + value_name),
+            "Value": (self.cli.option.VALUE, value),
+            "Type": (self.cli.option.VALUE, value_type),
+            "Action": (self.cli.option.VALUE, action),
+        }
+
+        self._add("Set-GPPrefRegistryValue", params)
+
+    def set_ssh(self, users: list[dict[str, bool, bool]]) -> None:
```

Comment:
Hahaha... Neither do I yet.  ;)

Reply:
Okay, I had to take some time and learn more Python, in particular classes, inheritance and instantiation to implement this.  So, this won't be used at all. The following "keys" that we need to edit, 'SeInteractiveLogon, SeRemoteInteractiveLogon, SeDenyInteractiveLogon, SeDenyRemoteInteractiveLogon' cannot be modified using PowerShell, because they are not in the registry. They are stored in the security policy database. This is why, I had to previously, backup the GPO, migrate it to the new domain, and import them. There is a lot of negatives implementing it this way but we may have to. 

There is something I am going to try first to avoid using templates, hopefully it'll work.